### PR TITLE
Update pr template and re-add Changelog builder

### PR DESCRIPTION
**Pull request by titu1994**
**Collection**: [docs]

# Changelog 
- Improves the PR template to include meta info + help reviewers in detailed review

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [x] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] Bugfix
- [x] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,15 +1,26 @@
-# What does this PR do?
+**Collection**: [Note which collection this PR will affect]
```
Comment (ericharper) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: I think we still need a header here, or people are going to miss filling out the collection.

Comment (titu1994) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: Ah good point.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,15 +1,26 @@
...
 - [ ] Did you write any new necessary tests?
-- [ ] Did you add or updated any necessary documentation?
-
+- [ ] Did you add or update any necessary documentation?
+- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
+  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
+  
+**PR Type**:
+- [ ] Bugfix
+- [ ] Documentation
```
[the remark this answers is not in the document]
Comment (titu1994) [reply .github/PULL_REQUEST_TEMPLATE.md:23]: New feature is the default and would almost always be selected though ?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (titu1994) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added a header.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (titu1994) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added it.

**Review by ericharper — APPROVED** [review]
LGTM. Thanks for the changes!

**Pull request merged**